monteCarloTreeSearch.c

```diff
@@ -0,0 +1,334 @@
+	int * action;				//action to be used for expansion
+	int * boardMove;            // boardMove for simulation
+
+	int randomValue;			//randomValue stores generated value used to pick moves for simulation
+	int timeElapsed;			//used when calculated time taken to complete a loop
+	int gameContinues;			//used to tell if game is still being simulated 
+	int numNewNodes = 0;		//keeps track of the number of nodes created
+	int winner;					//used for determining the winner
+	int numWins = 0; 			//used to tally the wins for backpropagation
+	int numSims = 0;			//used to tally simulations for back-propagation
+	int i;						//loop value
+	int j;						//loop value
+	int sum;					//used for summing the wins and simulations 
+  	int index;					//used for finding the index of the move to be made 
+  	int flag = 10;				//used if there are no moves to be made by the player
+	int numOfActions;			//used to track the number of moves available in simulation phase
+
+	double uctValue;			//used when calculating the node to expand
+	double bestValue;			//used to store the best value for expansion
+	clock_t startTime, timeCheck; //used to calculate the time that has passed
+
+	startTime = clock();				//track time limit
+	initializeRoot(root, gameBoard);	  // initialize root
+	newNode = root;  					// temporary newNode
+	selected = root;					//initialize selected value for expansion
+
+	while(numNewNodes < nodeLimit && flag > 0)
+	{
+		gameContinues = 100;	//initialize the simulation phase gameloop value
+		bestValue = -1000;		// initialize bestvalue as a negative number
+		numOfActions = 0;		//set numOfActions to zero each loop through
+
+		//1. Select a node using UCT and the functions provided by #1. (See the js implementation of 
+		//selection and the js UCT implementation as a guide).
+
+		//UCT calculation to select next node 
+		for(i = 0; i < newNode->numChildren; i++)
+		{
+			child = newNode->children[i];
+			//UCT calculation
+			uctValue = child->numWins / child->numSimulations + 
+			sqrt(2 * log(newNode->numSimulations / child->numSimulations));
+
+			//check if the uct value for each child is better than the one before
+			if(uctValue > bestValue)
+			{
+				selected = child;
+				bestValue = uctValue;
+			}
+		} 
+
+
+		
+		getMoves(gameBoard, allActions);
+		//2. Expand the selected node using the expand() function (see #1) after choosing a move. 
+		//It can be any unplayed move given by getMoves() (see #2).
+		for(i = 0; i < 200; i++)
+		{
+			if(allActions[i] == -1)
+			{
+				break;
+			}
+
+			numOfActions++;
+		}
+		if(numOfActions != 0)
+		{
+
+			for(i = 0; i < numOfActions; i+2)
+			{
+			
+				action[0] = allActions[i];
+				action[1] = allActions[i+ 1];
+				//You need to implement a choose move here and then send that move as action.
+				expand(selected, action);
+				numNewNodes++;
+
+			}
+		}
+		// else
+		// {
+		// 	move[0] = -1;
+		// 	move[1] = -1;
+		// 	flag == -100;
```

Comment:
I commented it out when I was unsure if I needed it there. I reorganized my code so that I did not need to use the flag there, but I left it there just in case